client/src/pages/ProfileListings/DateSelectPopover.tsx

```diff
@@ -0,0 +1,135 @@
+import CardHeader from '@material-ui/core/CardHeader';
+import CardActions from '@material-ui/core/CardActions';
+import Popover from '@material-ui/core/Popover';
+import CloseIcon from '@material-ui/icons/Close';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
+import DateFnsUtils from '@date-io/date-fns';
+import useStyles from './useStyles';
+import Alert from './alert';
+
+interface DateSelectProps {
+  open: true | false;
+  handleOpen: any;
+  handleUpdate: any;
+}
+
+const DateSelectPopover: React.FC<DateSelectProps> = ({ open, handleOpen, handleUpdate }) => {
+  const classes = useStyles();
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [dateFrom, setDateFrom] = useState<Date | null>(null);
+  const [dateTo, setDateTo] = useState<Date | null>(null);
+
+  const handleDateFromChange = (date: Date | null) => {
+    setDateFrom(date);
+  };
+
+  const handleDateToChange = (date: Date | null) => {
+    if (date && dateFrom && date < dateFrom) {
+      setSnackbarOpen(true);
+    } else {
+      setDateTo(date);
+    }
+  };
+
+  const updateDateRange = (dateFrom: Date | null, dateTo: Date | null) => {
+    dateFrom && dateTo ? handleUpdate(dateFrom, dateTo) : handleUpdate(null);
+  };
+
+  const handleSnackbarClose = (event?: React.SyntheticEvent, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  return (
+    <Popover
+      open={open}
+      className={classes.popover}
+      anchorOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+    >
+      <Card className={classes.datePickerCard}>
+        <CardHeader
+          className={classes.datePickerHeader}
+          action={
+            <IconButton aria-label="settings">
+              <CloseIcon onClick={handleOpen} />
+            </IconButton>
+          }
+        />
+        <Typography variant="h5" className={classes.datePickerTitle}>
+          Select date range
+        </Typography>
+        <MuiPickersUtilsProvider utils={DateFnsUtils}>
+          <KeyboardDatePicker
+            className={classes.datePicker}
+            disableToolbar
+            format="MM/dd/yyyy"
+            margin="normal"
+            id="date-picker-dialog"
+            label="From"
+            value={dateFrom instanceof Date ? dateFrom : null}
+            onChange={handleDateFromChange}
+            KeyboardButtonProps={{
+              'aria-label': 'change date',
+            }}
+          />
+          <KeyboardDatePicker
+            className={classes.datePicker}
+            disableToolbar
+            format="MM/dd/yyyy"
+            margin="normal"
+            id="date-picker-dialog"
```

Comment:
make sure you have different `id` for both these `<KeyboardDatePicker/>` component as under the hood there `id` is associated with the label. [https://www.w3.org/WAI/test-evaluate/preliminary/#labelschecks](https://www.w3.org/WAI/test-evaluate/preliminary/#labelschecks)